api_app/mixins.py

```diff
@@ -66,3 +75,516 @@ def list(self, request, *args, **kwargs):
                 cache.touch(cache_name, timeout=60 * 60 * 24 * 7)
 
         return Response(data)
+
+
+class VirusTotalv3BaseMixin(BaseAnalyzerMixin, metaclass=abc.ABCMeta):
```

Comment:
FYI if you made changes here I cannot view them cause the code has moved. If you have anything in particular to ask me to review please do it explicitly. Otherwise np

Reply:
I tried to generalize the mixin to make it extensible by both analyzers and ingestors.
About the parameter `url_sub_path` it should be **optional** but it was used directly, so I put a default so in the case of ingestors they don't have to worry about that value that they don't use.
I created the functions `_vt_download_file`, `_vt_intelligence_search` and `_vt_get_iocs_from_file` directly in the mixin so that both analyzers and ingestors can use them.

currently I'm missing:
- reduce the complexity of the function `_vt_get_report`
- do the tests of the functions in the mixin
- add more documentation? (in addition to the one I've already added here https://github.com/intelowlproject/docs/blob/main/docs/IntelOwl/usage.md#ingestors)

let me know if i forgot anything, thanks!